posthog/models/alert.py

```diff
@@ -1,20 +1,211 @@
+from datetime import datetime, UTC, timedelta
+from typing import Any, Optional
+
 from django.db import models
+from django.core.exceptions import ValidationError
+
+from posthog.hogql_queries.legacy_compatibility.flagged_conversion_manager import conversion_to_query_based
 from posthog.models.insight import Insight
+from posthog.models.utils import UUIDModel, CreatedMetaFields
+from posthog.schema import AlertCondition, InsightThreshold
 
 
 def are_alerts_supported_for_insight(insight: Insight) -> bool:
-    query = insight.query
-    if query is None or query.get("kind") != "TrendsQuery":
-        return False
-    if query.get("trendsFilter", {}).get("display") != "BoldNumber":
-        return False
+    with conversion_to_query_based(insight):
+        query = insight.query
+        if query is None or query.get("kind") != "TrendsQuery":
```

Comment:
It should also check if the query is `InsightVizNode`. It doesn't work for an insight with this example query:
```
{'kind': 'InsightVizNode', 'source': {'kind': 'TrendsQuery', 'series': [{'kind': 'EventsNode', 'math': 'total', 'name': '$pageview', 'event': '$pageview'}], 'trendsFilter': {'display': 'BoldNumber'}, 'breakdownFilter': None}}
```